### PR TITLE
fix: quantity on quickcheckout

### PR DESCRIPTION
**Pull request by rbedeoan-plenty**
[AB#107669](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/107669)

### CODE REVIEW

File: apps/web/components/QuickCheckout/QuickCheckout.vue [hunk 1]
```diff
@@ -30,9 +30,12 @@
...
 
-        <h1 class="font-bold typography-headline-4 mb-3" data-testid="product-name">
-          {{ productGetters.getName(product) }}
-        </h1>
+        <div class="flex">
+          <div class="mr-1 pt-1 font-normal typography-text-md">{{ quantity }} x</div>
+          <h1 class="font-bold typography-headline-4 mb-3" data-testid="product-name">
+            {{ productGetters.getName(product) }}
+          </h1>
+        </div>
```
Comment (abocsan-plenty) [inline apps/web/components/QuickCheckout/QuickCheckout.vue]:
```suggestion
        <div class="flex mb-3">
          <div class="mr-1 font-normal typography-text-md">{{ quantity }} x</div>
          <h1 class="font-bold typography-headline-4" data-testid="product-name">
            {{ productGetters.getName(product) }}
          </h1>
        </div>
```

File: apps/web/composables/useQuickCheckout/useQuickCheckout.ts [hunk 2]
```diff
@@ -58,8 +59,9 @@ export const useQuickCheckout: UseQuickCheckoutReturn = () => {
    * openQuickCheckout();
    * ```
    */
-  const openQuickCheckout: OpenQuickCheckout = (product: Product) => {
```
Comment (maxiroellplenty) [inline apps/web/composables/useQuickCheckout/useQuickCheckout.ts:61]: I dont get how this logic works, wouldnt this always override the quantity value. 
If I add a product with quantity 2 and then add the same product again with 2. The quantity value would show 2 and not 4 right?

Comment (maxiroellplenty) [reply apps/web/composables/useQuickCheckout/useQuickCheckout.ts:61]: Also, I would like to see a unit test for this behavior.

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

File: apps/web/composables/useQuickCheckout/useQuickCheckout.ts (hunk 2 above)
Comment (rbedeoan-plenty) [reply apps/web/composables/useQuickCheckout/useQuickCheckout.ts:61]: for what i understood overlay quantity should display last added quantity and product, but you ar right that it makes sens to take into consideration previous quantities

**Review by abocsan-plenty — DISMISSED** [review]

**Review by abocsan-plenty — DISMISSED** [review]

File: apps/web/composables/useQuickCheckout/__tests__/useQuickCheckout.spec.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { useQuickCheckout } from '../useQuickCheckout';
+import { ProductMock } from '../../../__tests__/__mocks__/product.mock';
+
+describe('useQuickCheckout', () => {
+  it('should check if open is true', () => {
+    const { openQuickCheckout, quantity, isOpen, product } = useQuickCheckout();
+    openQuickCheckout(ProductMock, 3);
+    expect(isOpen.value).toBe(true);
+    expect(product.value).toStrictEqual(ProductMock);
```
Comment (maxiroellplenty) [inline apps/web/composables/useQuickCheckout/__tests__/useQuickCheckout.spec.ts:14]: create own it for checking that the product is equal. always better to have only one expect in one it

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

**Review by maxiroellplenty — DISMISSED** [review]

**Review by maxiroellplenty — APPROVED** [review]

**Pull request merged**